evmbin/src/main.rs

```diff
@@ -405,15 +476,127 @@ mod tests {
+		let name = "add11".to_string();
+		let idx = 1;
+		// Simulate the chain `--chain CHAIN`
+		let spec = ForkSpec::EIP150;
+		let pre = _deserialized_state_tests.add11.pre_state.into();
+		let env_info = _deserialized_state_tests.add11.env.into();
+		let multitransaction = _deserialized_state_tests.add11.transaction;
+		for (spec, states) in _deserialized_state_tests.add11.post_states {
+			for (idx, state) in states.into_iter().enumerate() {
+				let informant = display::json::Informant::default();
+				// Hash of latest transaction index in the chain
+				let post_root = H256::from_str("99a450d8ce5b987a71346d8a0a1203711f770745c7ef326912e46761f14cd764").unwrap();
+				let trie_spec = TrieSpec::Secure; // TrieSpec::Fat for --std_dump_json
+				let transaction: transaction::SignedTransaction = multitransaction.select(&state.indexes).into();
+				info::run_transaction(&name, idx, &spec, &pre, post_root, &env_info, transaction, informant, trie_spec)
+			}
+		}
+	}
+
+	// Simulate using CLI command `state-test` and option `--json` (JSON informant)
+	// to execute a given transaction and verify its resulting state root
+	// using create2callPrecompiles.json
+	//
+	// ```
+	// cargo build -p evmbin;
+	// ./target/debug/parity-evm state-test \
+	//   ./evmbin/res/create2callPrecompiles.json --only create2callPrecompiles --chain Constantinople --json
+	// ```
+	fn should_verify_state_root_using_constantinople_state_test_json_file() {
+		let state_tests = include_str!("../res/create2callPrecompiles.json");
+		// Parse the specified state test JSON file to simulate the CLI command `state-test <file>`.
+		let _deserialized_state_tests: ConstantinopleStateTests = serde_json::from_str(state_tests)
+			.expect("Serialization cannot fail; qed");
+
+		// Simulate the name CLI option `--only NAME`
+		let name = "create2callPrecompiles".to_string();
+		let idx = 7;
+		// Simulate the chain `--chain CHAIN`
+		let spec = ForkSpec::Constantinople;
+		let pre = _deserialized_state_tests.create2callPrecompiles.pre_state.into();
+		let env_info = _deserialized_state_tests.create2callPrecompiles.env.into();
+		let multitransaction = _deserialized_state_tests.create2callPrecompiles.transaction;
+		for (spec, states) in _deserialized_state_tests.create2callPrecompiles.post_states {
+			for (idx, state) in states.into_iter().enumerate() {
+				let informant = display::json::Informant::default();
+				// Hash of latest transaction index in the chain
+				let post_root = H256::from_str("0xde1d3953b508913c6e3e9bd412cd50daf60bb177517e5d1e8ccb0dab193aed03").unwrap();
+				let trie_spec = TrieSpec::Secure; // TrieSpec::Fat for --std_dump_json
+				let transaction: transaction::SignedTransaction = multitransaction.select(&state.indexes).into();
+				info::run_transaction(&name, idx, &spec, &pre, post_root, &env_info, transaction, informant, trie_spec)
+			}
+		}
+	}
+
+	// TODO - add test that passes without failing with `State root mismatch`
```

Comment:
Please create an issue for those TODOs and add the issue number here.

Reply:
I've pushed commit b9db81e634fc6a3ba5159102f5ca1df07341f64a to provide a single TODO with a link to a new issue https://github.com/paritytech/parity-ethereum/issues/10768